hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMDatanodeProtocolServer.java

```diff
@@ -407,14 +387,48 @@ private static String flatten(String input) {
         .replaceAll(" +", " ");
   }
 
-  protected String getScmDatanodeAddressKey() {
+  protected RPC.Server getRpcServer(OzoneConfiguration conf,
+      InetSocketAddress datanodeRpcAddr,
```

Comment:
NIT: If we add a parameter Class<?> protocol for protocol? we don't need to overwrite getRpcServer in ReconDatanodeProtocolServer class.

Reply:
@xiaoyuyao I have refactored this a little bit differently. Please let me know if this is OK.